src/shell.ts

```diff
@@ -1,475 +1,150 @@
-import cp from "child_process"
+import { ChildProcess, spawn } from "child_process"
+import { randomUUID } from "crypto"
 import fs from "fs"
 import path from "path"
 import { promisify } from "util"
 
-import { CommandExecutor, CommandOutput, Context, ToString } from "./types"
-import { displayCommand, intoError, redact } from "./utils"
+import { Logger } from "./logger"
+import { Context, ToString } from "./types"
+import { displayCommand, redact } from "./utils"
 
 export const fsExists = promisify(fs.exists)
 export const fsReadFile = promisify(fs.readFile)
 export const fsWriteFile = promisify(fs.writeFile)
-const fsRmdir = promisify(fs.rmdir)
 const fsMkdir = promisify(fs.mkdir)
 const fsUnlink = promisify(fs.unlink)
-const cpExec = promisify(cp.exec)
 
-enum RetryContext {
-  CompilationError = "compilation error",
+export const ensureDir = async (dir: string) => {
+  if (!(await fsExists(dir))) {
+    await fsMkdir(dir, { recursive: true })
+  }
+  return dir
 }
-class Retry {
-  context: RetryContext
-  motive: string
-  stderr: string
 
-  constructor(options: {
-    context: RetryContext
-    motive: string
-    stderr: string
-  }) {
-    this.context = options.context
-    this.motive = options.motive
-    this.stderr = options.stderr
+export const initDatabaseDir = async (dir: string) => {
+  dir = await ensureDir(dir)
+  const lockPath = path.join(dir, "LOCK")
+  if (await fsExists(lockPath)) {
+    await fsUnlink(lockPath)
```

Comment:
And... basically the same as above

If you need to remove a file only if it exists, use `fs.rm` with `force` flag